pkg/provider/azure.go

```diff
@@ -1318,6 +1332,32 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 	}
 }
 
+// updateNodeTaint updates node out-of-service taint
+func (az *Cloud) updateNodeTaint(node *v1.Node) {
+	if node == nil {
+		return
+	}
```

Comment:
let's add `if az.KubeClient != nil` in the beginning, that would make the logic more clear since `if az.KubeClient == nil`, remove taint is impossible.

```
if az.KubeClient == nil {
  print warning msg
  return
}

Reply:
Update, thanks